spec/lib/shanty_docker_plugin/docker_plugin_spec.rb

```diff
@@ -0,0 +1,103 @@
+require 'shanty_docker_plugin/docker_plugin'
+require 'shanty_docker_plugin/image_wrapper'
+require 'shanty_docker_plugin/git_tag'
+require 'spec_helper'
+
+RSpec.describe(DockerPlugin::DockerPlugin) do
+  include_context('plugin')
+
+  it('adds the docker tag automatically') do
+    expect(described_class.tags).to match_array([:docker])
+  end
+
+  it('adds config for the output dir') do
+    expect(described_class).to add_config(:output_dir, 'build')
+  end
+
+  it('adds config for the fall back tag') do
+    expect(described_class).to add_config(:fallback_tag, 'latest')
+  end
+
+  it('adds config for the registry') do
+    expect(described_class).to add_config(:registry)
+  end
+
+  it('adds config for an optional tag') do
+    expect(described_class).to add_config(:tag)
+  end
+
+  it('subscribes to the build event') do
+    expect(described_class).to subscribe_to(:build).with(:on_build)
+  end
+
+  it('finds projects by calling a method to locate the ones that have a docker file') do
+    expect(described_class).to provide_projects(:resolve_projects)
+  end
+
+  let(:image_wrapper) { double('image_wrapper') }
+
+  before(:each) do
+    File.write(File.join(project_path, 'Dockerfile'), 'FROM ubuntu:15.10')
+    allow(project).to receive(:path).and_return(project_path)
+    allow(project).to receive(:parents).and_return([])
+    allow(project).to receive(:config).and_return(docker_project_name: 'one')
+    allow(project).to receive(:changed?).and_return(true)
+    allow(project).to receive(:all_artifacts).and_return([])
+    allow(DockerPlugin::GitTag).to receive(:current_branch).and_return('master')
+    allow(image_wrapper).to receive(:registry).and_return('niccage')
+  end
+
+  describe('.resolve_projects') do
+    it('find docker projects in the workspace and create dependencies between them') do
+      allow(file_tree).to receive(:glob).and_return([File.join(project_path, 'Dockerfile'),
+                                                     File.join(project_paths[1], 'Dockerfile'),
+                                                     File.join(project_paths[2], 'Dockerfile')])
+      File.write(File.join(project_paths[1], 'Dockerfile'), 'FROM <%= registry%>/one:master')
+      File.write(File.join(project_paths[2], 'Dockerfile'), 'FROM ubuntu:15.10')
```

Comment:
Use `project_paths.first` here instead of `project_path` - I would like to get rid of that `let` at some point as it makes it look like it's a special case when it's not.